fiona/ogrext1.pyx

```diff
@@ -210,6 +214,11 @@ cdef class FeatureBuilder:
                 else:
                     props[key] = datetime.datetime(
                         y, m, d, hh, mm, ss).isoformat()
+
+            elif fieldtype is bytes:
+                data = ogrext1.OGR_F_GetFieldAsBinary(feature, i, &l)
+                props[key] = PyBytes_FromStringAndSize(<char*>data, <Py_ssize_t>l)
```

Comment:
@snorfalorpagus let's let Cython handle the bytes object creation for us, like

```python
blob = data[:l]
props[key] = blob
```

See http://cython.readthedocs.io/en/latest/src/tutorial/strings.html#passing-byte-strings.